### ISSUE TITLE
Usuario expulsado 2 minutos antes de que terminara su reserva

### ISSUE THREAD

**Issue by angelagonzalez** [GLORIA-project/night-experiment-interface#26]
27-06-2014, 21:45 UT (reserva 5647):

GLORIA expulsó al usuario de la reserva a las 21:58 UT (2 minutos antes de lo debido), estando incluso todavía guardando las imágenes que había tomado durante la observación.

No se trata de un desfase entre la hora de GLORIA y la del ordenador. Se midió ese desfase al principio de la reserva, ya que está entera grabada (era para un tutorial), y el desfase era de tan sólo 2 segundos, como prueba la secuencia de imágenes que adjunto al final.

![27-06-2014 expulsin de la reserva 2 minutos antes](https://cloud.githubusercontent.com/assets/6812954/3421271/e9968222-fee2-11e3-849a-3ff2fd83996f.jpg)

![27-06-2014 expulsin 2 minutos antes desfase 2s](https://cloud.githubusercontent.com/assets/6812954/3421322/d8a37e2c-fee5-11e3-9853-1c028572407b.jpg)

**Comment by angelagonzalez** [comment]
Ha vuelto a ocurrir (28/06/2014 21:30UT, reserva 5850): el usuario dice: “it expelled me before my time had expired”.

**Comment by fserena** [comment]
La reserva 5850 terminó a las 21:45:00.812:

```
21:45:00.812 INFO  MountTeleoperation - {"action":{"rt":"BOOTES-1A","name":"ParkOperation","args":["BOOTES-1A","B1A"]},"sender":"teleoperation","client":"experiment"}
```

GLORIA solicitó el aparcado de la montura, como está programado para que haga siempre al final de las reservas. La primera solicitud del usuario al experimento tras la finalización de la reserva fue la lectura del estado de la montura (automática, la hace la interfaz periódicamente):

```
21:45:04.255 ERROR Experiment - {"action":{"name":"load_mount_status","rid":5850},"admin mode":false,"type":"ONLINE","name":"ExperimentOperation","rid":5850,"sender":"experiment","client":"*****"}
```

Después, hubo varias más hasta las  21:47:32:

```
21:47:32.605 ERROR Experiment - {"action":{"name":"load_mount_status","rid":5850},"admin mode":false,"type":"ONLINE","name":"ExperimentOperation","rid":5850,"sender":"experiment","client":"****"}
```

Esto tiene pinta de un desfase horario o de que la interfaz falló y no _echó_ al usuario correctamente al finalizar la reserva. Suponiendo que lo último funciona siempre (no hay incidencias de que el aviso no aparezca), yo me inclino más por la primera.

No conozco la forma en que esta interfaz gestiona el tiempo de reserva restante. Si lo hace descontando tiempo hasta la hora (absoluta) de fin de la reserva, el problema del desfase ocurrirá siempre que el usuario no tenga el reloj sincronizado con GLORIA. Si lo hace como la interfaz del experimento solar, no habría ningún problema: GLORIA sirve el número de segundos que le quedan al usuario, de forma que nunca hay confusión; cuando dicho valor llega a 0, se acabó (tenga la hora que tenga el usuario).

**Comment by angelagonzalez** [comment]
Tengo vídeo grabado, el reloj de nuestro ordenador (el viernes 27 Jun) iba 2 segundos desfasado con el de GLORIA, y la expulsión se produjo 2 minutos antes de tiempo. Así que¿podría estar ocurriendo que el reloj interno de GLORIA adelante? También le ha ocurrido a un usuario el sábado (28 Jun).

**Comment by esgg** [comment]
La interfaz no hace un uso directo del reloj de la máquina del usuario. Os comento como funciona:
1º/ Cuando el usuario accede a la interfaz, esta pide el tiempo que resta de experimento a los servicios.
2º/ La interfaz arranca un temporizador con la duración marcada por los usuarios.
3º/ Una vez cumplido el plazo, el temporizador pregunta cada x segundos (no me acuerdo ahora) a los servicios si el experimento ha concluido. Si contesta afirmativamente, lo expulsa.

El usuario ejecutó un f5 a las 21:42, probablemente al ser expulsado. Sabes si pudo acceder a la interfaz? La ejecución del remaining fue posterior a esta hora (21:43).

No veo ninguna irregularidad en los logs. Lo más probable es que sea por la interfaz pero no encuentro ninguna llamada a la función de tiempo restante antes de que se produjese el fallo

**Comment by carlibiri** [comment]
Esto es sencillo, si el reloj del equipo no está sincronizado con el del servidor de Gloria, cosa que es imposible, ocurrirá esto, pero no es un fallo, es que los dos relojes no están sincronizados. Si quiere comprobarlo, cuando entres en la reserva mira a ver cuánto tiempo te queda para acceder, y sino la otra opción es que utilices un cronometro.

Ángela si estás de acuerdo con eso puedes cerrar la incidencia.

Un saludo.

**Comment by angelagonzalez** [comment]
No, no se puede cerrar la incidencia. Existe un problema y no está resuelto. La comprobación de sincronizaciòn está hecha. Por favor leed la incidencia tal y como la reporto completa. El reloj de GLORIA y el de mi ordenador iban sincronizados con 2 segundos de desfase. La hora de windows es la oficial, se actualiza automáticamente por estar conectado a Internet. Mirad las capturas de pantalla. Me ocurrió a mí con testigos delante que tenían otros ordenadores con la misma hora que yo y quedaron igual de sorprendidos.  Le ocurrió a un usuario ajeno a GLORIA al día siguiente. Si el sistema está permitiendo acceder antes de la hora o no lo desconozco, peo si este es el caso hay que solucionarlo porque es un error.

**Comment by carlibiri** [comment]
Esteban, podéis mirar el reloj del servidor qué hora tiene? Un saludo.

**Comment by esgg** [comment]
Desde el control panel no puedo ver la hora. Pero conociendo a Alejandro es posible que haga la sincronización con algún servidor de hora externo. Se lo comento y os lo confirmo.

Ángela, ¿pudiste volver a acceder a la interfaz después de que te expulsara?

**Comment by angelagonzalez** [comment]
No lo intenté.

**Comment by esgg** [comment]
Voy a intentar reproducir el problema. Si te vuelve a pasar, prueba a entrar de nuevo, please. Es para descartar posibles causas.

**Comment by angelagonzalez** [comment]
De acuerdo, a ver esta noche si tengo ocasión.

**Comment by angelagonzalez** [comment]
Hoy no se puede probar, BOOTES-2 está cerrado.

**Comment by angelagonzalez** [comment]
Al final abrió BOOTES-2.

Reserva 6014: 01/07/2014 a las 22:15 UT. El sistema volvió a expulsarme cuando quedaba aproximadamente 1min 30 seg de tiempo. Probado con otro ordenador diferente, comprobada la hora también con el móvil. 

Sucedió igual que el viernes 27-06-2014: 
1º. Aparece el aviso de que quedan 2 minutos.
2º. Guardo una imagen.
3º. Al salir de la pantalla de guardado de imágenes, me expulsa de la reserva.

Esto no sucede al guardar imágenes cuando faltan más de 2 minutos y todavía no ha salido el aviso. Tampoco te expulsa si no guardas imágenes en el lapso comprendido entre 2 minutos y cero.

4º. Volví a intentar entrar en la reserva y sí pude. De nuevo apareció el aviso de "quedan 2 minutos".
5º.  Volví a guardar imágenes y esta vez no me expulsó.
6º. Esperé hasta que el tiempo llegó a cero.
7º. Antes de expulsarme de la reserva, estuvo en 00:00 durante 17 segundos.

Al igual que el viernes, todo está grabado porque es para el video tutorial.

**Comment by esgg** [comment]
Estoy intentando reproducir el fallo de la reserva pero no lo consigo. La CCD de B2 no funciona.

He probado con otros telescopios: TADn y B1A y el error no aparece. ¿Podrías indicarme que navegador y S.O estás utilizando?

Otra cosa, cuando te expulsó. ¿Te aparecían los enlaces de la encuesta?

**Comment by angelagonzalez** [comment]
Sí, aparecían los enlaces de la encuesta. Windows professional y Firefox 30. Con el otro ordenador Windows 7 home y Firefox 30. Me ocurrió en Bootes2 el viernes y también anoche.  Le ocurrió a un usuario también el sábado con D50. Desconozco su navegador y SO. La CCD de Bootes2 anoche funcionaba. Quizá está todo deshabilitado por ser de día. Te recomienda probar está noche si puedes. Saludos.

**Comment by angelagonzalez** [comment]
Windows 7 en ambos casos.

**Comment by angelagonzalez** [comment]
Si te ayuda puedo intentar enviarte el vídeo de la parte final.

**Comment by esgg** [comment]
Si, please

**Comment by esgg** [comment]
He probado con un windows y la última versión de Firefox y me funciona. Intentaré probar con B2 cuando la CCD esté habilitada. Me parece raro que el problema sea debido al tipo de telescopio. Sigo probando.

**Comment by angelagonzalez** [comment]
Te he enviado el vídeo por e-mail.

**Comment by esgg** [comment]
Acabo de verlo, gracias. La verdad es que es extraño porque no sucede siempre. Voy a seguir investigando

**Comment by angelagonzalez** [comment]
Anoche me volvió a ocurrir.

**Comment by esgg** [comment]
He desplegado una nueva versión para intentar arreglar este error. No he podido comprobarlo porque la CCD de B2 no ha estado disponible y no he podido repetir el experimento bajo las mismas condiciones.
